### PR TITLE
[Rule Tuning] High Number of Okta User Password Reset or Unlock Attempts

### LINKED ISSUE #1172: [Rule Tuning] [Okta] High Number of Okta User Password Reset or Unlock Attempts
## Link to rule
https://github.com/elastic/detection-rules/blob/main/rules/okta/defense_evasion_suspicious_okta_user_password_reset_or_unlock_attempts.toml

## Description
The above rule uses the wrong field for the threshold metric. Currently the threshold metric is ```okta.actor.id```, however, this value can sometimes be ```Okta System```. This results in two issues:

1. The rule can be triggered more frequently than intended, as ```Okta System``` can interact with multiple accounts within the time frame.
2. The alert doesn't contain any real useful information, as its been triggered off of the ```Okta System``` account, requiring additional investigation on the matter.

A third issue that is more about ```okta.actor.id``` in general:

3. ```okta.actor.id```  is just a random string, and doesn't provide any quick (at a glance) info, on who the user is, resulting in additional digging to find out if this is valid.

## Example Data

I think it would make more sense to use the threshold based off of something like okta.target.alternate_id as it solves the above problems. (One thing I'm not 100% sure on is whether this field is present during all events).

Example (truncated and snipped) event where okta.actor.id is of ```Okta System```

```
{
  "fields": {
    "okta.client.device": [
      "Mobile"
    ],
    "okta.client.ip": [
      "<snipped>"
    ],
    "service.type": [
      "okta"
    ],
    "okta.security_context.as.number": [
      <snipped>
    ],
    "user.full_name.text": [
      "Okta System"
    ],
    "event.outcome": [
      "success"
    ],
    "okta.uuid": [
      "<snipped>"
    ],
    "source.user.full_name.text": [
      "Okta System"
    ],
    "fileset.name": [
      "system"
    ],
    "input.type": [
      "httpjson"
    ],
    "okta.authentication_context.authentication_step": [
      0
    ],
    "related.user": [
      "Okta System"
    ],
    "okta.client.zone": [
      "null"
    ],
    "okta.target.type": [
      "User",
      "MobilePhone"
    ],
    "okta.authentication_context.external_session_id": [
      "unknown"
    ],
    "okta.security_context.is_proxy": [
      false
    ],
    "okta.actor.type": [
      "SystemPrincipal"
    ],
    "okta.debug_context.debug_data.url": [
      "/api/v1/authn/recovery/password?"
    ],
    "okta.actor.display_name": [
      "Okta System"
    ],
    "event.action": [
      "system.sms.send_password_reset_message"
    ],
    "okta.security_context.isp": [
      "<snipped>"
    ],
    "okta.outcome.result": [
      "SUCCESS"
    ],
    "okta.debug_context.debug_data.request_uri": [
      "/api/v1/authn/recovery/password"
    ],
    "okta.display_message": [
      "Send self-service password reset SMS message"
    ],
    "client.user.full_name": [
      "Okta System"
    ],
    "okta.target.id": [
      "<snipped (real user)>",
      "<snipped (real user)>"
    ],
    "okta.actor.alternate_id": [
      "system@okta.com"
    ],
    "event.kind": [
      "event"
    ],
    "client.user.id": [
      "<snipped>"
    ],
    "okta.security_context.domain": [
      "<snipped>"
    ],
    "okta.client.user_agent.browser": [
      "<snipped>"
    ],
    "okta.transaction.id": [
      "<snipped>"
    ],
    "ecs.version": [
      "1.8.0"
    ],
    "okta.target.display_name": [
      "<snipped> (real user)",
      "<snipped> (real user)"
    ],
    "user.full_name": [
      "Okta System"
    ],
    "agent.version": [
      "7.12.1"
    ],
    "okta.debug_context.debug_data.request_id": [
      "<snipped>"
    ],
    "source.user.full_name": [
      "Okta System"
    ],
    "okta.target.alternate_id": [
      "<snipped> (real user)",
      "<snipped> (real user)"
    ],
    "okta.event_type": [
      "system.sms.send_password_reset_message"
    ],
    "okta.debug_context.debug_data.threat_suspected": [
      "false"
    ],
    "okta.transaction.type": [
      "WEB"
    ],
    "event.module": [
      "okta"
    ],
    "okta.actor.id": [
      "<snipped> (okta system)"
    ],
    "source.user.id": [
      "<snipped> (okta system)"
    ],
    "client.user.full_name.text": [
      "Okta System"
    ],
    "okta.security_context.as.organization.name": [
      "<snipped>"
    ],
    "event.dataset": [
      "okta.system"
    ]
  }
}
```

### PR DESCRIPTION
**Pull request by austinsonger**
## Issues

#Resolves #1172 

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

**Comment by threat-punter** [comment]
Nice work @austinsonger and @BenB196!

@BenB196 would you like your first and last name listed under the `author` field in the rule?

**Comment by austinsonger** [comment]
@threat-punter on #1172 @BenB196 told me he only wanted `@`

**Comment by BenB196** [comment]
Yeah, my ```@``` is fine for now as the author.

File: rules/okta/defense_evasion_suspicious_okta_user_password_reset_or_unlock_attempts.toml [hunk 1]
```diff
@@ -82,6 +82,6 @@ name = "Initial Access"
 reference = "https://attack.mitre.org/tactics/TA0001/"
 
 [rule.threshold]
-field = ["okta.actor.id"]
+field = ["okta.actor.alternate_id"]
```
Comment (threat-punter) [inline rules/okta/defense_evasion_suspicious_okta_user_password_reset_or_unlock_attempts.toml:85]: I've seen this change between `okta.target.alternate_id` and `okta.actor.alternate_id`. Does anyone have an environment up and running to test this out and post the redacted event?

Comment (BenB196) [reply rules/okta/defense_evasion_suspicious_okta_user_password_reset_or_unlock_attempts.toml:85]: So it depends on the event type:

For password resets/unlocks, both ```okta.target.alternate_id``` and ```okta.actor.alternate_id``` exist.

Here are a snipped few examples:

<details>
<summary>event.action: "system.sms.send_account_unlock_message"</summary>

Of this action's events:

- ```okta.actor.alternate_id``` always is ```system@okta.com```
- ```okta.target.alternate_id``` contains both ```<snipped (real user)>``` and ```<snipped (real user's phone number)>```

```json
{
  "_source" : {
    "event" : {
      "kind" : "event",
      "module" : "okta",
      "action" : "system.sms.send_account_unlock_message",
      "dataset" : "okta.system",
      "outcome" : "success"
    },
    "okta" : {
      "actor" : {
        "id" : "<snipped>",
        "display_name" : "Okta System",
        "type" : "SystemPrincipal",
        "alternate_id" : "system@okta.com"
      },
      "debug_context" : {
        "<snipped>": "<snipped>"
      },
      "event_type" : "system.sms.send_account_unlock_message",
      "authentication_context" : {
        "authentication_step" : 0,
        "external_session_id" : "unknown"
      },
      "display_message" : "Send self-service account unlock SMS message",
      "client" : {
        "<snipped>": "<snipped>"
      },
      "uuid" : "<snipped>",
      "outcome" : {
        "result" : "SUCCESS"
      },
      "transaction" : {
        "id" : "<snipped>",
        "type" : "WEB"
      },
      "security_context" : {
        "<snipped>": "<snipped>"
      },
      "target" : [
        {
          "id" : "<snipped (real user)>",
          "type" : "User",
          "display_name" : "<snipped (real user)>",
          "alternate_id" : "<snipped (real user)>"
        },
        {
          "id" : "<snipped (real user's phone number)>",
          "type" : "MobilePhone",
          "display_name" : "<snipped (real user's phone number)>",
          "alternate_id" : "<snipped (real user's phone number)>"
        }
      ]
    }
  }
}
```
</details>

<details>
<summary>event.action: "system.sms.send_password_reset_message"</summary>

Of this action's events:

- ```okta.actor.alternate_id``` always is ```system@okta.com```, with the exception of 1 event which has the value of ```<snipped (real user)>```
- ```okta.target.alternate_id``` contains both ```<snipped (real user)>``` and ```<snipped (real user's phone number)>```

```json
{
  "_source" : {
    "event" : {
      "kind" : "event",
      "module" : "okta",
      "action" : "system.sms.send_password_reset_message",
      "dataset" : "okta.system",
      "outcome" : "success"
    },
    "okta" : {
      "actor" : {
        "id" : "<snipped>",
        "display_name" : "Okta System",
        "type" : "SystemPrincipal",
        "alternate_id" : "system@okta.com"
      },
      "debug_context" : {
        "<snipped>" : "<snipped>"
      },
      "event_type" : "system.sms.send_password_reset_message",
      "authentication_context" : {
        "authentication_step" : 0,
        "external_session_id" : "unknown"
      },
      "display_message" : "Send self-service password reset SMS message",
      "client" : {
        "<snipped>" : "<snipped>"
      },
      "uuid" : "<snipped>",
      "outcome" : {
        "result" : "SUCCESS"
      },
      "transaction" : {
        "id" : "<snipped>",
        "type" : "WEB"
      },
      "security_context" : {
        "<snipped>" : "<snipped>"
      },
      "target" : [
        {
          "id" : "<snipped (real user)>",
          "type" : "User",
          "display_name" : "<snipped (real user)>",
          "alternate_id" : "<snipped (real user)>"
        },
        {
          "id" : "<snipped (real user's phone number)>",
          "type" : "MobilePhone",
          "display_name" : "<snipped (real user's phone number)>",
          "alternate_id" : "<snipped (real user's phone number)>"
        }
      ]
    }
  }
}
```
</details>

<details>
<summary>event.action: "system.voice.send_account_unlock_call"</summary>

Of this action's events:

- ```okta.actor.alternate_id``` always is ```system@okta.com```
- ```okta.target.alternate_id``` contains both ```<snipped (real user)>``` and ```<snipped (real user's phone number)>```

```json
{
  "_source" : {
    "event" : {
      "kind" : "event",
      "module" : "okta",
      "action" : "system.voice.send_account_unlock_call",
      "dataset" : "okta.system",
      "outcome" : "success"
    },
    "okta" : {
      "actor" : {
        "id" : "<snipped>",
        "display_name" : "Okta System",
        "type" : "SystemPrincipal",
        "alternate_id" : "system@okta.com"
      },
      "debug_context" : {
        "<snipped>" : "<snipped>"
      },
      "event_type" : "system.voice.send_account_unlock_call",
      "authentication_context" : {
        "authentication_step" : 0,
        "external_session_id" : "unknown"
      },
      "display_message" : "Send self-service account unlock call",
      "client" : {
        "<snipped>" : "<snipped>"
      },
      "uuid" : "<snipped>",
      "outcome" : {
        "result" : "SUCCESS"
      },
      "transaction" : {
        "id" : "<snipped>",
        "type" : "WEB"
      },
      "security_context" : {
        "<snipped>" : "<snipped>"
      },
      "target" : [
        {
          "id" : "<snipped (real user)>",
          "type" : "User",
          "display_name" : "<snipped (real user)>",
          "alternate_id" : "<snipped (real user)>"
        },
        {
          "id" : "<snipped (real user's phone number)>",
          "type" : "MobilePhone",
          "display_name" : "<snipped (real user's phone number)>",
          "alternate_id" : "<snipped (real user's phone number)>"
        }
      ]
    }
  }
}
```
</details>

<details>
<summary>event.action: "system.voice.send_password_reset_call"</summary>

Of this action's events:

- ```okta.actor.alternate_id``` always is ```system@okta.com```
- ```okta.target.alternate_id``` contains both ```<snipped (real user)>``` and ```<snipped (real user's phone number)>```

```json
{
  "_source" : {
    "event" : {
      "kind" : "event",
      "module" : "okta",
      "action" : "system.voice.send_password_reset_call",
      "dataset" : "okta.system",
      "outcome" : "success"
    },
    "okta" : {
      "actor" : {
        "id" : "<snipped>",
        "display_name" : "Okta System",
        "type" : "SystemPrincipal",
        "alternate_id" : "system@okta.com"
      },
      "debug_context" : {
        "<snipped>" : "<snipped>"
      },
      "event_type" : "system.voice.send_password_reset_call",
      "authentication_context" : {
        "authentication_step" : 0,
        "external_session_id" : "unknown"
      },
      "display_message" : "Send self-service password reset call",
      "client" : {
        "<snipped>" : "<snipped>"
      },
      "uuid" : "<snipped>",
      "outcome" : {
        "result" : "SUCCESS"
      },
      "transaction" : {
        "id" : "<snipped>",
        "type" : "WEB"
      },
      "security_context" : {
        "<snipped>" : "<snipped>"
      },
      "target" : [
        {
          "id" : "<snipped (real user)>",
          "type" : "User",
          "display_name" : "<snipped (real user)>",
          "alternate_id" : "<snipped (real user)>"
        },
        {
          "id" : "<snipped (real user's phone number)>",
          "type" : "MobilePhone",
          "display_name" : "<snipped (real user's phone number)>",
          "alternate_id" : "<snipped (real user's phone number)>"
        }
      ]
    }
  }
}
```
</details>

<details>
<summary>event.action: "user.account.unlock_token"</summary>

Of this action's events:

- ```okta.actor.alternate_id``` always is ```system@okta.com```
- ```okta.target.alternate_id``` contains ```<snipped (real user)>```

```json
{
  "_source" : {
    "event" : {
      "kind" : "event",
      "module" : "okta",
      "action" : "user.account.unlock_token",
      "dataset" : "okta.system",
      "outcome" : "success"
    },
    "okta" : {
      "actor" : {
        "id" : "<snipped>",
        "display_name" : "Okta System",
        "type" : "SystemPrincipal",
        "alternate_id" : "system@okta.com"
      },
      "debug_context" : {
        "<snipped>" : "<snipped>"
      },
      "event_type" : "user.account.unlock_token",
      "authentication_context" : {
        "authentication_step" : 0,
        "external_session_id" : "unknown"
      },
      "display_message" : "Issued recovery token for self-service account unlock",
      "client" : {
        "<snipped>" : "<snipped>"
      },
      "uuid" : "<snipped>",
      "outcome" : {
        "result" : "SUCCESS",
        "reason" : "Issued recovery token"
      },
      "transaction" : {
        "id" : "<snipped>",
        "type" : "WEB"
      },
      "security_context" : {
        "<snipped>" : "<snipped>"
      },
      "target" : [
        {
          "type" : "User",
          "alternate_id" : "<snipped (real user)>"
        }
      ]
    }
  }
}
```
</details>

Based off the updated wording of the PR, I believe it is correct that ```okta.actor.alternate_id``` is being used here, as it's looking for a high number of resets/unlocks across the environment(s).


Note: I don't have any examples of event.action ```system.email.password_reset.sent_message``` or ```system.email.account_unlock.sent_message``` currently.

Comment (threat-punter) [reply rules/okta/defense_evasion_suspicious_okta_user_password_reset_or_unlock_attempts.toml:85]: Thanks for the detailed reply!

**Review by threat-punter — COMMENTED** [review]
Nice one!

**Review by threat-punter — APPROVED** [review]

**Comment by bm11100** [comment]
@austinsonger can you re-push the update from main and see if it passes unit tests, seems like some were skipped? then we can merge this PR.

**Review by bm11100 — APPROVED** [review]

**Pull request merged**